internal/config/image/cuda_image.go

```diff
@@ -271,6 +284,28 @@ func (i CUDA) DevicesFromMounts() []string {
 	return devices
 }
 
+func (i CUDA) VisibleDevices() []string {
+	// If enabled, try and get the device list from volume mounts first
+	if i.acceptDeviceListAsVolumeMounts {
+		devices := i.visibleDevicesFromMounts()
+		if len(devices) > 0 {
+			return devices
+		}
+	}
+
+	// Fallback to reading from the environment variable if privileges are correct
+	devices := i.VisibleDevicesFromEnvVar()
+	if len(devices) == 0 {
+		return nil
+	}
+
+	if i.isPrivileged || i.acceptEnvvarUnprivileged {
+		return devices
+	}
+
```

Comment:
The one thing that we're missing from the initial implementation is that we're not logging the fact that we're ignoring the devices requested as envvars.

Let's add a logger to the `CUDA` image as well and add the logging here:
```
	i.logger.Warningf("Ignoring devices specified in NVIDIA_VISIBLE_DEVICES in unprivileged container")
```